ovirt-imageio.spec.in

```diff
@@ -220,6 +240,51 @@ Python client library for accessing imageio server on oVirt hosts.
 
 %endif
 
+%if %{with_python311}
+%define python3_pkgversion 3.11
+%define __python3 /usr/bin/python3.11
+
+%package -n python3.11-%{name}-common
+Summary:   oVirt imageio common resources
+
+BuildRequires: gcc
+BuildRequires: python3.11-devel
+
+Requires:  python3.11
+
+%description -n python3.11-%{name}-common
+Common resources used by oVirt imageio server and client
+
+%files -n python3.11-%{name}-common
+%license LICENSES/GPL-2.0-or-later.txt
+%{python3_sitearch}/%{srcname}
+%{python3_sitearch}/%{srcname}-*.egg-info
+%exclude %{python3_sitearch}/%{srcname}/client
+%exclude %{python3_sitearch}/%{srcname}/admin
+
+
+%package -n python3.11-%{name}-client
+Summary:   oVirt imageio client library
+
+Requires:  python3.11-%{name}-common = %{version}-%{release}
+
+%if 0%{?rhel}
+# RHEL 8.4 version. Some features require qemu-nbd 5.2.0 and are disabled when
+# using older qemu-nbd.
+Requires:  qemu-img >= 15:4.2.0
+%else
+# Fedora.
+Requires:  qemu-img
+%endif
+
+%description -n python3.11-%{name}-client
+Python client library for accessing imageio server on oVirt hosts.
+
+%files -n python3.11-%{name}-client
+%{python3_sitearch}/%{srcname}/client
+
+%endif
```

Comment:
This block does not make sense. You don't need to duplicate so much,
the macro with_python311 should be enouth to modify the package name
and whatever needed.

I see we already have this duplication for python 3.9, this is bad.

Please refactor the spec to remove the duplication for python 3.9
before adding python 3.11.

Ideally `%package common` can be parametrized so it can create python3.9-{%name}-common and python3.11-{%name}-common, or we would have one copy of this package that can be paramertized to create python{%python_version}-{%name}-common.

Reply:
Updated, tested on stream 9 and checked out the artefacts stream 8 also worked.